Logs on storage by default (#167)
NethServer/nethsecurity-docs: storage.rst

Storage
=======

> Starting from version 8.6, NethSecurity automatically ensures that system logs are saved to a persistent storage partition.  

Reviewer: Logs are not persistent across reboots on installations on virtual machines without additional storage.
We might say "tries to ensure...if storage is available".

This guarantees that logs remain available across reboots or unexpected shutdowns, even if the storage has not been manually configured.
The default log retention period is 52 weeks.

For **new installations** the system automatically creates a dedicated partition on the main disk to store logs.  

For **upgrades**:

* if no storage had previously been configured, the system automatically sets it up using unallocated space on the primary disk.  

> * If storage was already configured, it remains unchanged

Reviewer 2: ```suggestion
* if storage was already configured, it remains unchanged
```

.. note::

   This behavior improves reliability and does not require manual intervention. However, users can still manage storage settings from the web interface.  

Persistent log storage can be disabled (not recommended), or moved to a different disk if needed.  
If disabled, the system will automatically reconfigure it during a future upgrade.

Manual configuration
^^^^^^^^^^^^^^^^^^^^

Manual configuration of additional storage is still available and works as follows:

* If using an additional device, connect it to the system.
* Access the ``Storage`` page under the ``System`` section in the right-hand menu.
* Select the storage device where logs should be saved.
* Click on the button :guilabel:`Format and configure storage`.
* If the selected device is the **primary disk**, the system will generate a new partition using any unallocated space.
* If an **additional disk** is selected, the system will erase all existing partitions and data and create a single new partition.

The storage is then:

* Formatted with the ``ext4`` filesystem
* Mounted at ``/mnt/data``
* Used by ``rsyslog`` to write logs to ``/mnt/data/log/messages``
* Rotated weekly by ``logrotate``

> * Synced daily (at night) for additional data like metrics

Reviewer 2: ```suggestion
* Synchronized daily (at night) for additional data like metrics
```

To remove persistent storage and return to in-memory logging, click on the button :guilabel:`Remove storage`.

Virtual Machines
----------------
When installing NethSecurity on a virtual machine, the recommended method is to generate the virtual disk from the official image.
In this mode, logs are not stored persistently by default.

> To enable persistent log storage, you must attach a second virtual disk to the virtual machine.
> The automatic internal partitioning method is not supported on virtual disks.

Reviewer 2: ```suggestion
To enable persistent log storage, you must attach a second virtual disk to the virtual machine.
As an alternative, you can extend the virtual disk and use the free disk space to create a new partition like on a physical hardware.
```

Behavior in versions prior to 8.6
---------------------------------

> In earlier versions of NethSecurity, logs were written by default to a **volatile in-memory directory** to avoid stressing the root filesystem in case of failure.
> 
> * Memory-based storage had **limited capacity**
> * Old logs were automatically removed as space filled up
> * **All logs were lost** after a reboot or system failure
> 
> To persist logs, storage had to be configured **manually**, either by using unallocated space on the system disk or by attaching a secondary disk.  
> This step was necessary to avoid log loss, but required explicit action by the user.

Reviewer 2: System is still more reliable without a persistent log, but we are enabling it to ease the life of the support team.
So it's better to just describe the old behavior without more explanation.

```suggestion
In earlier versions of NethSecurity, logs were written by default to a **volatile in-memory directory**.
To persist logs, storage had to be configured **manually**, either by using unallocated space on the system disk or by attaching a secondary disk.  
```
